packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -432,22 +445,15 @@ contract Deploy is Deployer {
             _finalizationPeriodSeconds: cfg.finalizationPeriodSeconds()
         });
 
-        require(oracle.SUBMISSION_INTERVAL() == cfg.l2OutputOracleSubmissionInterval());
-        require(oracle.submissionInterval() == cfg.l2OutputOracleSubmissionInterval());
-        require(oracle.L2_BLOCK_TIME() == cfg.l2BlockTime());
-        require(oracle.l2BlockTime() == cfg.l2BlockTime());
-        require(oracle.PROPOSER() == cfg.l2OutputOracleProposer());
-        require(oracle.proposer() == cfg.l2OutputOracleProposer());
-        require(oracle.CHALLENGER() == cfg.l2OutputOracleChallenger());
-        require(oracle.challenger() == cfg.l2OutputOracleChallenger());
-        require(oracle.FINALIZATION_PERIOD_SECONDS() == cfg.finalizationPeriodSeconds());
-        require(oracle.finalizationPeriodSeconds() == cfg.finalizationPeriodSeconds());
-        require(oracle.startingBlockNumber() == 0);
-        require(oracle.startingTimestamp() == 0);
-
         save("L2OutputOracle", address(oracle));
         console.log("L2OutputOracle deployed at %s", address(oracle));
 
+        Types.ContractSet memory contracts = _proxiesUnstrict();
```

Comment:
The outer function here is deploying a the L2OutputOracle's implementation contract, does it make sense that we're getting the proxies?

Reply:
In the L2OO's case, no, because it doesn't rely on any other contracts in the set for assertions. To keep things generic enough for reuse between the impl / proxy upgrade steps, these `checkXYZ` functions accept a contract set, and in the impl deployments, we manually override the contract that was deployed within the set to the impl address.

This is done, for ex, for the `OptimismPortal`, where on deployment we set the `L2_ORACLE` immutable to the `L2OutputOracleProxy`. In the assertions within `checkOptimismPortal`, we need to check that the `L2_ORACLE` address is what we expected it to be, and that address exists within the proxy contract set.

Reply:
or something like `_getContracts(bool _isProxies)`

Reply:
That's kind of the weird part - the only contract that we want to be an impl in the sets we retrieve within the `deployXYZ` functions is the contract that was deployed. All other contracts should be proxies, or `address(0)` if they're not available, due to the way we instantiate references to other protocol contracts in the impl deployments (they're always the proxies).

Reply:
So for ex, in the `OptimismPortal` deployment, we grab the proxy set (which contains the `L2OutputOracleProxy`, which we need to perform the check), but replace the `OptimismPortal` address w/ the impl.